build.gradle

```diff
@@ -458,3 +458,22 @@ project(':iceberg-spark-runtime') {
   }
 }
 
+String getProjectVersion() {
+  try {
+    return gitVersion()
+  } catch (NullPointerException e) {
+    def noGitVersion = '0.1.0-SNAPSHOT'
```

Comment:
Can this version be read from a file instead of defaulted? A source tarball release won't include a .git directory, but needs to build the right version. We could add a file to the root that specifies the version that this reads to set the version correctly.

Reply:
> I think we might also want to build with version from version.txt sometimes even if it's regular git clone

That sounds like a good idea to me.

Reply:
Added project property "useStableVersion" to enable building project with "./gradlew assemble -PuseStableVersion=true" with version from .txt file.

Please let me know if you any suggestion on this approach.

Reply:
I think we should use the contents of `version.txt` if the file exists, and the git version otherwise. That way, the same build command works.

Reply:
It makes sense and I updated to use contents of version.txt first. Thanks for suggestion!